lib/pipeline.js

```diff
@@ -799,11 +800,33 @@ class PipelineModel extends BaseModel {
         /* eslint-enable global-require */
         const factory = SecretFactory.getInstance();
 
-        const secrets = factory.list(listConfig);
+        let secrets = factory.list(listConfig);
+
+        // Fetch and merge config pipeline's secrets
+        if (this.configPipelineId && this.configPipelineId !== this.id) {
+            const configPipelineListConfig = hoek.clone(listConfig);
+
+            configPipelineListConfig.params.pipelineId = this.configPipelineId;
+
+            return Promise.all([
+                secrets,
+                factory.list(configPipelineListConfig)
+            ]).then((results) => {
+                // Merge config pipeline's secrets into this pipeline's
+                secrets = _.uniqBy([...results[0], ...results[1]], 'name');
```

Comment:
Don't think uniqBy is what we want based on this example:
```
_.uniqBy([{ 'x': 1 }, { 'x': 2 }, { 'x': 1 }], 'x');
// => [{ 'x': 1 }, { 'x': 2 }]
```
https://lodash.com/docs#uniqBy
Can probably use `hoek.applyToDefaults` instead. https://github.com/hapijs/hoek/blob/master/API.md#applytodefaultsdefaults-options-isnulloverride

Reply:
I may be wrong, but if the goal is to override any secrets with the same name, then I think `_.uniqBy` is the more appropriate approach.

For example, given these array of secret objects:
```
const list1 = [{name: 'secret1'}, {name: 'secret2'}];
const list2 = [{name: 'secret3'}, {name: 'secret4'}];
```

We expect the resulting list to include all of the secret objects since they all have unique names.

Output of `_.uniqBy([...list1, ...list2], 'name');`:
```
[ { name: 'secret1' },
  { name: 'secret2' },
  { name: 'secret3' },
  { name: 'secret4' } ]
``` 

Output of `hoek.applyToDefaults(list1, list2);`:
```
[ { name: 'secret3' }, { name: 'secret4' } ]
```

Reply:
loll I get the secrets model wrong! yep, _uniq is the right way to go!